src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryFormHelper.ts

```diff
@@ -0,0 +1,202 @@
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+import * as Yup from 'yup'
+import { cloneDeep, unset, size, isUndefined } from 'lodash-es'
+import { TerraformStoreTypes } from '../TerraformInterfaces'
+
+export const formatInitialValues = (isConfig: boolean, prevStepData: any, isTerraformPlan: boolean) => {
+  const defaultArtifacts = [
+    {
+      artifactFile: {
+        artifactPathExpression: undefined,
+        name: undefined,
+        path: undefined
+      }
+    }
+  ]
+  if (isConfig && isTerraformPlan) {
+    return {
+      spec: {
+        configuration: {
+          configFiles: {
+            store: {
+              spec: {
+                repositoryName:
+                  prevStepData.formValues?.spec?.configuration?.configFiles?.store.spec?.repositoryName || '',
+                artifacts:
+                  prevStepData.formValues?.spec?.configuration?.configFiles?.store.spec?.artifacts || defaultArtifacts
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  if (isConfig) {
+    return {
+      spec: {
+        configuration: {
+          configFiles: {
+            store: {
+              spec: {
+                repositoryName:
+                  prevStepData.formValues?.spec?.configuration?.spec?.configFiles?.store.spec?.repositoryName || '',
+                artifacts:
+                  prevStepData.formValues?.spec?.configuration?.spec?.configFiles?.store.spec?.artifacts ||
+                  defaultArtifacts
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  return {
+    varFile: {
+      identifier: prevStepData?.varFile?.identifier || '',
+      type: TerraformStoreTypes.Remote,
+      spec: {
+        store: {
+          spec: {
+            repositoryName: prevStepData?.varFile?.spec?.store?.spec?.repositoryName || '',
+            artifacts: prevStepData?.varFile?.spec?.store?.spec?.artifacts || defaultArtifacts
+          }
+        }
+      }
+    }
+  }
+}
+
+export const getConnectorRef = (isConfig: boolean, isTerraformPlan: boolean, prevStepData: any) => {
+  let connectorValue
+  if (isConfig && isTerraformPlan) {
+    connectorValue = prevStepData?.formValues.spec?.configuration?.configFiles?.store?.spec?.connectorRef
+  } else if (isConfig) {
+    connectorValue = prevStepData?.formValues?.spec?.configuration?.spec?.configFiles?.store?.spec?.connectorRef
+  } else connectorValue = prevStepData?.varFile?.spec?.store?.spec?.connectorRef
+
+  if (size(prevStepData?.identifier) > 5) {
+    return prevStepData?.identifier
+  }
+
+  if (!isUndefined(connectorValue?.connector?.identifier)) {
+    return connectorValue.connector.identifier
+  }
+
+  return connectorValue
+}
+
+export const terraformArtifactorySchema = (isConfig: boolean, getString: any) => {
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232248)
```suggestion
export const terraformArtifactorySchema = (isConfig: boolean, getString: unknown) => {
```